SpaceGame/Assets/Scripts/MotherShipCollisionHandler.cs

```diff
@@ -79,10 +79,21 @@ public void OnTriggerEnter(Collider other)
 
     public void Notify()
     {
+
+        IObserver expiredObserver = null;
         foreach (IObserver observer in m_Observers)
         {
-            observer.GetUpdate(this);
+            if (observer != null)
+            {
+                observer.GetUpdate(this);
+            }
+            //remvoe null observers
+            else
```

Comment:
this check is bogus, you assing the observer to be removed from the list if is is null, yet below you explicitly check for it not to be null ?